File: src/test/testdata/inspections/unsafeIterators/IdentityMap.java
package main;

import com.badlogic.gdx.utils.IdentityMap;

import java.util.Iterator;

class IdentityMapTest {

  private IdentityMap<String, Object> map1 = new IdentityMap<String, Object>();

  void test() {

    for (String s: <warning>map1.keys()</warning>) {
      System.out.println(s);
    }

    <warning>map1.keys()</warning>.iterator();

    Iterator iterator1 = <warning>map1.iterator()</warning>;
    Iterator iterator2 = <warning>map1.values()</warning>;
    IdentityMap.Keys<String> iterator3 = <warning>map1.keys()</warning>;
    Iterator iterator4 = <warning>map1.entries()</warning>;

    for (IdentityMap.Entry e: <warning>map1</warning>) {
      System.out.println(e.key);
    }

  }

}

File: src/test/testdata/inspections/unsafeIterators/IdentityMap.kt
package main

import com.badlogic.gdx.utils.IdentityMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST")
class IdentityMapTest {

  var map1 = IdentityMap<String, Any>()

  fun test() {

    for (s in <warning>map1.keys()</warning>) {
      println(s)
    }

    <warning>map1.keys()</warning>.iterator()

    val iterator1 = <warning>map1.iterator()</warning>
    val iterator2 = <warning>map1.values()</warning>
    val iterator3 = <warning>map1.keys()</warning>
    val iterator4 = <warning>map1.entries()</warning>

    for (e in <warning>map1</warning>) {
      println(e.key)
    }

    <warning>(if (true) IdentityMap<String, String>() else IdentityMap<String, Any>()).keys()</warning>

  }

}
